### PR TITLE
Fix `latest-tag-button` at repo home on tagged default branch

### LINKED ISSUE #5446: `latest-tag-button` is showing "+0" and "HEAD isn't tagged"
### Please ensure:

- [X] The bug is caused by Refined GitHub. It doesn't happen if I disable the extension.
- [X] The bug happens after clearing extension cache. The "Clear cache" button below can also be found at the bottom of the options page.

### Example URLs

https://github.com/kas-elvirov/gloc

### Extension version

22.2.13

### Browser(s) used

Safari 15.3

### Description

I'm opening an issue to track https://github.com/refined-github/refined-github/pull/5365#issuecomment-1047870611

Screen Shot 11

HEAD is tagged:

Screen Shot 12

### Include in this issue:

- [X] Screenshots/video/gif demonstrating the bug, if it’s visual
- [ ] Console errors, if any

### PR DESCRIPTION
**Pull request by cheap-glitch**
Fixes #5446 

## Test URLs

- [Latest tag](https://github.com/refined-github/refined-github/tree/22.3.3)
- [Default branch, tagged](https://github.com/kas-elvirov/gloc)
- [Default branch, not tagged](https://github.com/refined-github/github-url-detection)
- [Other branches](https://github.com/refined-github/refined-github/tree/hotfix)

## Screenshot

**Before**
before

**After**
after

### CODE REVIEW

File: source/features/latest-tag-button.tsx [hunk 1]
```diff
@@ -117,7 +117,8 @@ async function init(): Promise<false | void> {
 
 	const defaultBranch = await getDefaultBranch();
 
-	if (currentBranch === latestTag || (currentBranch === defaultBranch && aheadBy === 0)) {
+	// `getCurrentCommittish` returns `undefined` when at the repo root on the default branch #5446
+	if (currentBranch === latestTag || ((!currentBranch || currentBranch === defaultBranch) && aheadBy === 0)) {
```
Comment (fregante) [inline source/features/latest-tag-button.tsx:121]: Can we split this into 2/3 variables? e.g. `onLatestTag` _ `onDefaultBranch` + `!isAhead`

**Review by fregante — APPROVED** [review]

**Comment by fregante** [comment]
I get "You're on the latest version" on this repo, with the latest version of the extension (not from the store)

Screen Shot

**Comment by cheap-glitch** [comment]
It works fine on my side:

image

@fregante did you clear the cache?

**Comment by fregante** [comment]
Yes I have

**Comment by cheap-glitch** [comment]
> with the latest version of the extension

Actually, I just noticed this PR hasn't been released yet. My screenshot above was with the dev version on latest main commit.

**Comment by fregante** [comment]
I mean locally built, **not from the store**, what's on `main` currently

**Pull request merged**